### ISSUE TITLE
Add dedicate network and static IP for podman driver

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#9714]
in minikube 1.14.0 we had docker driver with dedicate network, we should do same thing for podman for consistancy

- Similar to this PR 
https://github.com/kubernetes/minikube/pull/9294

- This will make Podman driver to also have a Static IP similar to our Docker Driver after that PR
- You would need to install Podman on Linux to test this

**Comment by afbjorklund** [comment]
The podman network doesn't have label nor options, that might become a problem...


```
Usage:	docker network create [OPTIONS] NETWORK

Create a network

Options:
      --attachable           Enable manual container attachment
      --aux-address map      Auxiliary IPv4 or IPv6 addresses used by Network driver (default map[])
      --config-from string   The network from which copying the configuration
      --config-only          Create a configuration only network
  -d, --driver string        Driver to manage the Network (default "bridge")
      --gateway strings      IPv4 or IPv6 Gateway for the master subnet
      --ingress              Create swarm routing-mesh network
      --internal             Restrict external access to the network
      --ip-range strings     Allocate container ip from a sub-range
      --ipam-driver string   IP Address Management Driver (default "default")
      --ipam-opt map         Set IPAM driver specific options (default map[])
      --ipv6                 Enable IPv6 networking
      --label list           Set metadata on a network
  -o, --opt map              Set driver specific options (default map[])
      --scope string         Control the network's scope
      --subnet strings       Subnet in CIDR format that represents a network segment
```


```
Usage:
  podman network create [flags] [NETWORK]

Examples:
  podman network create podman1

Flags:
      --disable-dns      disable dns plugin
  -d, --driver string    driver to manage the network (default "bridge")
      --gateway ip       IPv4 or IPv6 gateway for the subnet
      --internal         restrict external access from this network
      --ip-range ipNet   allocate container IP from range
      --macvlan string   create a Macvlan connection based on this device
      --subnet ipNet     subnet in CIDR format
```

Also "docker" is hardcoded in a dozen place in the old PR, instead of using ociBin.

**Comment by afbjorklund** [comment]
See https://github.com/containers/podman/issues/7989#issuecomment-706552196, it needs a podman network rewrite first (before labels)

I'm not sure if any of the options were mandatory, or if there is some kind of CNI workaround ?
`--ip-masq`
`--icc`

https://docs.docker.com/engine/extend/plugins_network/
https://github.com/moby/libnetwork/tree/master/drivers/bridge/

```
Usage: bridge [ OPTIONS ] OBJECT { COMMAND | help }
       bridge [ -force ] -batch filename
where	OBJECT := { link | fdb | mdb | vlan | monitor }
	OPTIONS := { -V[ersion] | -s[tatistics] | -d[etails] |
		     -o[neline] | -t[imestamp] | -n[etns] name |
		     -c[ompressvlans] -color -p[retty] -j[son] }
```

**Comment by afbjorklund** [comment]
The filter label functionality seems broken anyway, so most likely it needs manual cleanup...

Just like #9705 for the volumes ?

**Comment by afbjorklund** [comment]
Podman network labels were discussed here: https://github.com/containers/podman/issues/5376

It _might_ become available in Podman 3.0, along with the fix for --filter ...
